### PR TITLE
fix false memory toleration calculation when loading segments on qnode

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #23262

### CODE REVIEW

**Comment by MrPresent-Han** [comment]
rerun ut

File: internal/querynode/segment_loader.go [hunk 1]
```diff
@@ -637,12 +637,13 @@ func GetStorageSizeByIndexInfo(indexInfo *querypb.FieldIndexInfo) (uint64, uint6
 
 func (loader *segmentLoader) checkSegmentSize(collectionID UniqueID, segmentLoadInfos []*querypb.SegmentLoadInfo, concurrency int) error {
 	usedMem := hardware.GetUsedMemoryCount()
-	totalMem := hardware.GetMemoryCount()
+	currentAvailableMemCount := hardware.GetAvailableMemoryCount()
+	totalMem := currentAvailableMemCount + usedMem
```
Comment (congqixia) [inline internal/querynode/segment_loader.go:641]: NIT: please add some comment here to explain why total memory cannot be used

**Review by congqixia — APPROVED** [review]
we need some test for this in container env with specified limit and request
/lgtm

**Comment by xiaofan-luan** [comment]
I don't think we can simply go with request.
If you limits at 9G, they are actually expecting the memory can be used up to 8G or something.
If the pod is killed due to lack of memory on one of the pod that looks ok to me, we just need to check with @zwd1208 to make sure it will be reassigned to a pod with 9G memory or more.

**Comment by xiaofan-luan** [comment]
I do see an issue in gc_tuner.go, where we always use the pod limit to estimated next GOGC target to improve performance when free memory is large, and reduce mem usage by set GOGC to small values and garbage collect earlier.

What is the good memory limit to use? Request? Limit?

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by congqixia** [comment]
@xiaofan-luan I think @MrPresent-Han discussed with you offline about the this policy
Manually add ci-passed due to code coverage.

**Pull request merged**